hoodie-client/src/main/java/com/uber/hoodie/index/InMemoryHashIndex.java

```diff
@@ -43,26 +43,26 @@
 
   private static ConcurrentMap<HoodieKey, HoodieRecordLocation> recordLocationMap;
 
-  public InMemoryHashIndex(HoodieWriteConfig config, JavaSparkContext jsc) {
-    super(config, jsc);
+  public InMemoryHashIndex(HoodieWriteConfig config) {
+    super(config);
     recordLocationMap = new ConcurrentHashMap<>();
   }
 
   @Override
   public JavaPairRDD<HoodieKey, Optional<String>> fetchRecordLocation(JavaRDD<HoodieKey> hoodieKeys,
-      final HoodieTable<T> table) {
+      JavaSparkContext jsc, HoodieTableMetaClient metaClient) {
```

Comment:
again, why remove table argument?

Reply:
Index is moved inside table, don't think it makes sense for index to pass table into it. WDYT ?

Reply:
Left a comment above.. I think we should pass the table if we can.